docs/src/kubernetes_upgrade.md

```diff
@@ -1,54 +1,105 @@
-3. re-joining the node to the cluster (`uncordon`)
+1. eviction of workloads (`drain`): workloads are gracefully moved away from
+   the node to be updated, ensuring a smooth transition.
+2. performing the operation: the actual maintenance operation, such as a
+   system update or hardware replacement, is executed.
+3. rejoining the node to the cluster (`uncordon`): the updated node is
+   reintegrated into the cluster, ready to resume its responsibilities.
 
-The above process requires workloads to be either stopped for the
-entire duration of the upgrade or migrated to another node.
+This process requires either stopping workloads for the entire upgrade duration
+or migrating them to other nodes in the cluster.
 
-While the latest case is the expected one in terms of service
-reliability and self-healing capabilities of Kubernetes, there can
-be situations where it is advised to operate with a temporarily
-degraded cluster and wait for the upgraded node to be up again.
+## Temporary PostgreSQL Cluster Degradation
 
-In particular, if your PostgreSQL cluster relies on **node-local storage**
-\- that is *storage which is local to the Kubernetes worker node where
-the PostgreSQL database is running*.
-Node-local storage (or simply *local storage*) is used to enhance performance.
+While the standard approach ensures service reliability and leverages
+Kubernetes' self-healing capabilities, there are scenarios where operating with
+a temporarily degraded cluster may be advisable. This is particularly relevant
+for PostgreSQL clusters relying on **node-local storage**, where the storage is
+local to the Kubernetes worker node running the PostgreSQL database. Node-local
+storage, or simply *local storage*, is employed to enhance performance.
 
 !!! Note
-    If your database files are on shared storage over the network,
-    you may not need to define a maintenance window. If the volumes currently
-    used by the pods can be reused by pods running on different nodes after
-    the drain, the default self-healing behavior of the operator will work
-    fine (you can then skip the rest of this section).
-
-When using local storage for PostgreSQL, you are advised to temporarily
-put the cluster in **maintenance mode** through the `nodeMaintenanceWindow`
-option to avoid standard self-healing procedures to kick in,
-while, for example, enlarging the partition on the physical node or
-updating the node itself.
+    If your database files reside on shared storage accessible over the
+    network, the default self-healing behavior of the operator can efficiently
+    handle scenarios where volumes are reused by pods on different nodes after a
+    drain operation. In such cases, you can skip the remaining sections of this
+    document.
+
+## Pod Disruption Budgets
+
+By default, CloudNativePG ensures that the primary node in a PostgreSQL cluster
+prohibits its own draining, safeguarding uninterrupted operations.
```

Comment:
IMO this is correct but possibly confusing. The user does have the ability to select a node for draining.
Suggestion
```suggestion
By default, CloudNativePG ensures that if a node to be drained contains the
cluster's primary instance, a switchover happens ahead of the drain. Once the
instance in the node is downgraded to replica, the draining can proceed.
For single-instance clusters, CloudNativePG will prevent draining the node where
the instance is housed, safeguarding operations.
```